Gems/Atom/Utils/Code/Source/AssetCollectionAsyncLoader.cpp

```diff
@@ -222,8 +224,8 @@ namespace AZ
 
         Data::AssetBus::MultiHandler::BusConnect(assetId);
         // Kick off asset loading.
-        auto asset = Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::PreLoad);
-        asset.BlockUntilLoadComplete();
```

Comment:
Since this was previously blocking, does something need to wait for the map to be empty after everything is queued?

Reply:
Not In this case. All the shared data is accessed (and protected) during OnAssetReady/OnAssetError.